ocppj/dispatcher.go

```diff
@@ -517,9 +517,8 @@ func (d *DefaultServerDispatcher) messagePump() {
 			}
 		case clientID = <-d.readyForDispatch:
 			// Client can now transmit again
-			rdy = true
+			clientQueue, rdy = d.queueMap.Get(clientID)
 			clientReadyMap[clientID] = rdy
```

Comment:
Nice catch. This leads to another follow-up edge case:
1. `ServerQueueMap.Remove(client)` is the primary cause
2. `delete(clientReadyMap, clientID)` is called inside the messagePump
3. `clientReadyMap[clientID] = rdy` -> re-adds a just deleted element. 

In theory not a problem, since the client will most likely reconnect again eventually, but it's a recipe for memory leaks. I would wrap it in a conditional:
```
if rdy {
  clientReadyMap[clientID] = rdy
}
```